lib/config/react.js

```diff
@@ -21,7 +21,6 @@ module.exports = {
 
   rules: merge(require('./rules/react'), require('./rules/jsx-a11y'), {
     'shopify/react-initialize-state': 'error',
-    'shopify/react-type-state': 'error',
```

Comment:
We no longer need this rule, as TS has updated to make it so the that `state` is correctly checked against its type argument. I guess I should just delete the rule altogether?

Reply:
👍 for 🔥 the rule.